pkg/comp-functions/functions/vshnmariadb/proxysql.go

```diff
@@ -0,0 +1,510 @@
+package vshnmariadb
+
+import (
+	"bytes"
+	"context"
+	"crypto/md5"
+	"encoding/hex"
+	"fmt"
+	"sort"
+	"text/template"
+
+	xfnproto "github.com/crossplane/function-sdk-go/proto/v1beta1"
+	"github.com/vshn/appcat/v4/apis/kubernetes/v1alpha2"
+	vshnv1 "github.com/vshn/appcat/v4/apis/vshn/v1"
+	"github.com/vshn/appcat/v4/pkg/comp-functions/runtime"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	pdbv1 "k8s.io/api/policy/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/ptr"
+)
+
+const (
+	configTmpl = `datadir="/var/lib/proxysql"
+
+    admin_variables=
+    {
+        admin_credentials="admin:{{ .RootPassword }};radmin:{{ .RootPassword }}"
+        mysql_ifaces="0.0.0.0:6032"
+        refresh_interval=2000
+        cluster_username="radmin"
+        cluster_password="{{ .RootPassword }}"
+    }
+
+    {{ if eq .TLS 1 }}
+    mysql_variables = {
+        ssl_p2s_ca = "/var/lib/proxysql/proxysql-ca.pem"
+    }
+    {{ end }}
+
+    mysql_servers =
+    (
+        { address="{{ .CompName }}-0.{{ .CompName }}-headless.{{ .Namespace }}.svc" , port=3306 , hostgroup=2, weight=120, use_ssl={{ .TLS }} },
+        { address="{{ .CompName }}-1.{{ .CompName }}-headless.{{ .Namespace }}.svc" , port=3306 , hostgroup=2, weight=110, use_ssl={{ .TLS }} },
+        { address="{{ .CompName }}-2.{{ .CompName }}-headless.{{ .Namespace }}.svc" , port=3306 , hostgroup=2, weight=100, use_ssl={{ .TLS }} }
+    )
+
+    mysql_users =
+    (
+        { username = "root", password = "{{ .RootPassword }}", default_hostgroup = 2 },
+        {{range $val := .Users}}
+        { username = "{{ $val.Name }}", password = "{{ $val.Password }}", default_hostgroup = 2 },
+        {{end}}
+    )
+
+    mysql_galera_hostgroups =
+    (
+        {writer_hostgroup=2,backup_writer_hostgroup=4,reader_hostgroup=3,offline_hostgroup=1,active=1,max_writers=1,writer_is_also_reader=1,max_transactions_behind=100}
+    )
+
+    proxysql_servers =
+    (
+        { hostname = "proxysql-0.proxysqlcluster.{{ .Namespace }}.svc", port = 6032, weight = 1 },
+        { hostname = "proxysql-1.proxysqlcluster.{{ .Namespace }}.svc", port = 6032, weight = 1 }
+    )`
+)
+
```

Comment:
can't we just move it to separate file and then template it with text/template? 
#readibility

Reply:
Good idea!